### DISCUSSION TITLE
issue: Type issues with string array when using useFieldArray hook

### DISCUSSION THREAD

**Discussion by milindgoel15** [react-hook-form/react-hook-form#11592] [Q&A]
### Version Number

^7.50.1

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/react-hook-form-usefieldarray-forked-2vtmkp?file=%2Fsrc%2Findex.tsx%3A10%2C45

### Steps to reproduce

1. Add default values in string array format
2. Notice it will throw type error

```
Type 'string' is not assignable to type 'never'.typescript(2322)
fieldArray.d.ts(6, 5): The expected type comes from property 'name' which is declared here on type 'UseFieldArrayProps<{ test: string[]; }, never, "id">'
```

### Expected behaviour

No type error. It should work normally with string based array as it does it custom object based array.

### What browsers are you seeing the problem on?

none, type error thrown in IDE

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by milindgoel15** [comment]
More details as the issue template didnt had any description field:

Just to note that on my code, it also throws type error on append() function
```
Argument of type 'string' is not assignable to parameter of type '{ name: string; } | Details | ({ name: string; } | Details)[]'
```

Moreover, I have also made a feature.tsx file with my code to reproduce the issue with my code. Although my code works normally except the type error. If required, I can setup a minimal reproducible repo with my code.

**Comment by Moshyfawn** [comment]
Your codesandbox is inaccessible.

**Comment by milindgoel15** [comment]
Didnt notice it was private. Changed visibility to public. Please lemme know what other details to provide.

**Comment by Moshyfawn** [comment]
From the [`useFieldArray` docs](https://react-hook-form.com/docs/usefieldarray):

> Does not support flat field array.

`useFieldArray` has to work with objects as opposed to simple types like `string`, because RHF has to keep track of each field by inserting an `id` property.

So the solution would be to convert `test: ["Bill", "John"]` to `test: [{ name: "Bill" }, { name: "John" }]`.

**Comment by milindgoel15** [comment]
But it appears to work even with simple type.

Edit: Although I can surely modify my useForm schema and map over the data array to extract the value in a separate array when saving to DB.
